### ISSUE TITLE
Explanation pre- and post-processing pipeline

### ISSUE THREAD

**Issue by aaarrti** [understandable-machine-intelligence-lab/Quantus#249]
### Description of the problem
- attribution preprocessing before metrics should be available as separate modules which could be pipelined, like torchvision.transform or sklearn Pipeline. 
- At the moment, attribution preprocessing, such as normalise or interpolation, and even - attribution computation like softmax, are currently performed under-the-hood in each metric.
- is a follow-up for https://github.com/understandable-machine-intelligence-lab/Quantus/issues/186
